src/apps/bridge/reportBuilder.cpp

```diff
@@ -484,19 +540,40 @@ static void report_builder_task(void *parameters) {
                   if (element == NULL) {
                     constexpr size_t bufsize = 80;
                     static char buffer[bufsize];
-                    int len = snprintf(buffer, bufsize,
-                                       "No data for node %" PRIx64
-                                       " in report period, adding it to the list\n",
-                                       _ctx._report_period_node_list[i]);
-                    BRIDGE_LOG_PRINTN(buffer, len);
-                    // we have a sensor that may have been added at the end of the report period
-                    // so lets add it to the list and this wil backfill all of the data so we can still send it
-                    // in the report. We will need to pass (_ctx._sample_counter - 1) to make sure we don't overflow the sensor_data buffer.
-                    // Also we will pass NULL into the sensor_data since we don't have any data for it yet and we will fill the whole thing with NANs
-                    _ctx._reportBuilderLinkedList.findElementAndAddSampleToElement(
-                        _ctx._report_period_node_list[i], AANDERAA_SENSOR_TYPE, NULL,
-                        sizeof(aanderaa_aggregations_t), _ctx._samplesPerReport,
-                        (_ctx._sample_counter - 1));
+                    if (_ctx._report_period_sensor_type_list[i] > SENSOR_TYPE_UNKNOWN) {
+                      int len = snprintf(buffer, bufsize,
+                                         "No data for node %" PRIx64
+                                         " in report period, adding it to the list\n",
+                                         _ctx._report_period_node_list[i]);
+                      BRIDGE_LOG_PRINTN(buffer, len);
+                      switch (_ctx._report_period_sensor_type_list[i]) {
+                      case AANDERAA_SENSOR_TYPE: {
+                        _ctx._reportBuilderLinkedList.findElementAndAddSampleToElement(
+                            _ctx._report_period_node_list[i],
+                            _ctx._report_period_sensor_type_list[i], NULL,
+                            sizeof(aanderaa_aggregations_t), _ctx._samplesPerReport,
+                            (_ctx._sample_counter - 1));
+                        break;
+                      }
+                      case SOFT_SENSOR_TYPE: {
+                        _ctx._reportBuilderLinkedList.findElementAndAddSampleToElement(
+                            _ctx._report_period_node_list[i],
+                            _ctx._report_period_sensor_type_list[i], NULL,
+                            sizeof(soft_aggregations_t), _ctx._samplesPerReport,
+                            (_ctx._sample_counter - 1));
+                        break;
+                      }
+                      default: {
+                        BRIDGE_LOG_PRINT("Invalid sensor type in report_builder_task\n");
+                        break;
+                      }
+                      }
```

Comment:
this is where we would add a sensor that is in our topology, but failed to be subscribed to by sensorController. The sensor type is required so we know what size of data we will be adding to the report.